src/tally.F90

```diff
@@ -845,6 +1032,122 @@ subroutine score_fission_eout(p, t, i_score)
   end subroutine score_fission_eout
 
 !===============================================================================
+! SCORE_FISSION_DELAYED_EOUT handles a special case where we need to store
+! delayed neutron production rate with an outgoing energy filter (think of a
+! fission matrix). In this case, we may need to score to multiple bins if there
+! were multiple neutrons produced with different energies.
+!===============================================================================
+
+  subroutine score_fission_delayed_eout(p, t, i_score)
+
+    type(Particle), intent(in)       :: p
+    type(TallyObject), intent(inout) :: t
+    integer, intent(in)              :: i_score ! index for score
+
+    integer :: i             ! index of outgoing energy filter
+    integer :: j             ! index of delayedgroup filter
+    integer :: d             ! delayed group
+    integer :: g             ! another delayed group
+    integer :: d_bin         ! delayed group bin index
+    integer :: n             ! number of energies on filter
+    integer :: k             ! loop index for bank sites
+    integer :: bin_energyout ! original outgoing energy bin
+    real(8) :: score         ! actual score
+    real(8) :: E_out         ! energy of fission bank site
+    logical :: d_found = .false. ! bool to inidicate if delayed group was found
```

Comment:
No need to initialize this variable. Also, keep in mind that when you initialize a variable where it is declared, the initialization is done only once at program startup (not every time the subroutine is entered).